### ISSUE TITLE
Flatpak build 

### ISSUE THREAD

**Issue by LeandroStanger** [LibreTranslate/LibreTranslate#438]
Please, consider supporting [Flatpak](http://docs.flatpak.org/en/latest/introduction.html#reasons-to-use-flatpak) and publishing **LibreTranslate** on [Flathub](https://github.com/flathub/flathub/wiki/App-Submission).

**Comment by pierotofy** [comment]
Hey @LeandroStanger :wave: would you be interested in contributing a PR that helps build/deploy a Flatpak package?

**Comment by vemonet** [comment]
Hi @LeandroStanger and @pierotofy (long time!)

As far as I know flatpak is built mainly to publish cross platform graphical desktop applications in a containerized and secure manner.

LibreTranslate is more of a server system: an endpoint that you deploy, and which can be accessed by any user on the network. Which is a completely different philosophy (and for this kind of system docker images are the best way to package and distribute imo)

To me it would feel a bit weird to start an app as a server with flatpak, and then access the app through localhost on my web browser. I would expect the app maintainers to have bundled the webapp in a desktop GUI.

That said, if the LibreTranslate maintainers would like to make it easier for users to directly install and use LibreTranslate as a desktop app, flatpak would be the best packaging option imo. From my experience building and maintaining a flatpak is not too hard: finding how to build all your dependencies the first time is a bit cumbersome, but then it is quite stable. It just adds a bit more of complexity to the codebase and release workflow, as it requires a separate repo under the flathub org containing 3 files with app metadata and build instructions. And to publish a new version on flathub the maintainers will need to change the build version in the flathub repo, and make a PR to the `master` branch

To distribute LibreTranslate as a desktop app are some frameworks for JS webapps such as Electron or Tauri. But not sure if it will be easy with the current LibreTranslate stack mostly in python

So, in conclusion:
* Do we want to push the use of LibreTranslate as a standalone desktop app? (precisely, do the maintainers want to support the extra work required to maintain this?)
* If yes, do we want to let it work as a server, or try to bundle it as a full blown desktop app?

**Comment by pierotofy** [comment]
I'd personally be open to have a desktop standalone app. Of course we'd need help maintaining / updating it.
